backend/apid/graphql/entity.go

```diff
@@ -79,6 +79,17 @@ func (r *entityImpl) Related(p schema.EntityRelatedFieldResolverParams) (interfa
 		return []*types.Entity{}, err
 	}
 
+	// omit source
+	for i, en := range entities {
+		if en.ID != entity.ID {
+			continue
+		}
+		entities[i] = entities[len(entities)-1]
+		entities[len(entities)-1] = nil
+		entities = entities[:len(entities)-1]
+		break
+	}
```

Comment:
> Why are we overwriting the indexed entity with the last entity? Why is entities being reduced by one element?

See the "Delete without preserving order" section of [Golang's Slice Trick](https://github.com/golang/go/wiki/SliceTricks#delete-without-preserving-order) document.

- In our case we can delete without preserving order because we sort in the next step.
- Since the slice is made up of pointers we explicitly set the final element of the slice to `nil` to ensure it is no longer referenced and can be GC'd.

Reply:
Thanks for the explanation and added comment in the code. Looks good!

LGTM!